### PR TITLE
tests: trickle: convert legacy test to ztest

### PR DESCRIPTION
**Pull request by punitvara**
This patch makes legacy test to use ztest APIs.

Signed-off-by: Punit Vara <punit.vara@intel.com>

### CODE REVIEW

File: tests/net/trickle/src/main.c [hunk 1]
```diff
@@ -206,103 +163,60 @@ static bool test_trickle_1_wait_long(void)
...
+			ztest_unit_test(test_trickle_1_status),
+			ztest_unit_test(test_trickle_2_status),
+			ztest_unit_test(test_trickle_1_wait),
+			ztest_unit_test(test_trickle_2_wait),
+			ztest_unit_test(test_trickle_1_update),
+			ztest_unit_test(test_trickle_1_status),
+			ztest_unit_test(test_trickle_2_inc),
+			ztest_unit_test(test_trickle_2_status),
+			ztest_unit_test(test_trickle_1_status),
+#if CHECK_LOG_TIMEOUT > 1
```
Comment (jukkar) [inline tests/net/trickle/src/main.c]: This should be `#if CHECK_LONG_TIMEOUT > 0` like in the original code.

**Review by jukkar — CHANGES_REQUESTED** [review]

File: tests/net/trickle/src/main.c (hunk 1 above)
Comment (punitvara) [reply tests/net/trickle/src/main.c]: @jukkar  Done.

File: tests/net/trickle/src/main.c [hunk 2]
```diff
@@ -206,103 +163,60 @@ static bool test_trickle_1_wait_long(void)
...
+			ztest_unit_test(test_trickle_1_status),
+			ztest_unit_test(test_trickle_2_status),
+			ztest_unit_test(test_trickle_1_wait),
+			ztest_unit_test(test_trickle_2_wait),
+			ztest_unit_test(test_trickle_1_update),
+			ztest_unit_test(test_trickle_1_status),
+			ztest_unit_test(test_trickle_2_inc),
+			ztest_unit_test(test_trickle_2_status),
+			ztest_unit_test(test_trickle_1_status),
+#if CHECK_LOG_TIMEOUT > 0
```
Comment (jukkar) [inline tests/net/trickle/src/main.c]: You have still typo here, it should be CHECK_LONG_TIMEOUT

**Review by jukkar — CHANGES_REQUESTED** [review]

File: tests/net/trickle/src/main.c (hunk 2 above)
Comment (punitvara) [reply tests/net/trickle/src/main.c]: Done

Comment (punitvara) [reply tests/net/trickle/src/main.c]: @jukkar Can you please review once. ?

Comment (jukkar) [reply tests/net/trickle/src/main.c]: I will after you have sent the fix.

Comment (punitvara) [reply tests/net/trickle/src/main.c]: Done. Sorry it was hanging locally. Forget to push

**Review by jukkar — APPROVED** [review]

**Pull request merged**